### PR TITLE
#2393 Document why Falco is written in C++ rather than anything else

### LINKED ISSUE #2393: Document why Falco is in C++ and unlikely to be ported
**Motivation**
Currently compiling go plugins for use with falco is complicated by the fact they must be compiled to a shared library, which brings the usual [downsides of dynamic linking](https://github.com/falcosecurity/falco-website/issues/801)

If Falco was entirely written in go (with certain parts as c libraries as required) we could remove that friction

Falco being written in C++ also makes it more difficult for Go developers to contribute to as it requires yet another toolchain

**Feature**

Falco is a pure go binary, and adding go plugins can do done without requiring cgo

**Alternatives**

**Additional context**
This is just a starting question, rather than a holy war!

### PR DESCRIPTION
**Pull request by RichardoC**
**What type of PR is this?**

 /kind documentation

**Any specific area of the project related to this PR?**

/area documentation

**What this PR does / why we need it**:
Capturing the details from <https://github.com/falcosecurity/falco/issues/2393> so that maintainers don't have to keep answering

**Which issue(s) this PR fixes**:

Fixes #2393

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RichardoC** [comment]
/remove-lifecycle stale
This is just waiting for a review

File: README.md [hunk 1]
```diff
@@ -159,6 +159,19 @@ Falco is licensed to you under the [Apache 2.0](./COPYING) open source license.
...
+4. As a security tool meant to consume a crazy high throughput of events per second, Falco needs to squeeze performance in all hot paths at runtime and requires deep control on memory allocation, which the Go runtime can't provide (there's also garbage collection involved).
+5. Although Go didn't suit the engineering requirements of the core of Falco, we still thought that it could be a good candidate for writing Falco extensions through the plugin system. This is the main reason we gave special attention and high priority to the development of the plugin-sdk-go.
+6. Go is not a requirement for having statically-linked binaries. In fact, we provide fully-static Falco builds since few years. The only issue with those is that the plugin system can't be supported with the current dynamic library model we currently have.
+7. Your opening point is very tied to plugins developed in Go, but we need to be mindful that the plugin system has been envisioned to support multiple languages too, so on our end maintaining a C-compatible codebase is the best strategy to ensure maximum cross-language compatibility.
```
Comment (FedeDP) [inline README.md:170]: > 7. Your opening point is very tied to plugins developed in Go, 

Which opening point?

**Review by FedeDP — COMMENTED** [review]
This LGTM; i just left a minor comment.

File: README.md (hunk 1 above)
Comment (FedeDP) [reply README.md:170]:
```suggestion
7. The plugin system has been envisioned to support multiple languages, so on our end maintaining a C-compatible codebase is the best strategy to ensure maximum cross-language compatibility.
```

**Comment by FedeDP** [comment]
Since this is a doc-only PR, i'll move to the next milestone.
We can merge this at any time!
/milestone 0.36.0

**Comment by RichardoC** [comment]
These build failures seem unrelated to this docs change?`sudo: ./unit_tests/falco_unit_tests: command not found`

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — DISMISSED** [review]
/approve

**Comment by Andreagit97** [comment]
ei @RichardoC we are ready to merge it but we need a rebase :/ if you want we can do it for you but we need to allow us enabling this
rebase

**Comment by RichardoC** [comment]
> ei @RichardoC we are ready to merge it but we need a rebase :/ if you want we can do it for you but we need to allow us enabling it in this PR: rebase

I've rebased

**Pull request merged**